src/common/utxobased/engine/utils.ts

```diff
@@ -35,6 +35,13 @@ export const getCurrencyFormatFromPurposeType = (
   }
 }
 
+export const getNumberOfBytesFromHex = (hex: string): number => {
```

Comment:
This is very wasteful.
If you're string is already a valid hex sting, then the num of bytes is just `Math.ceil(hex.length / 2)`